utils/misc.ml

```diff
@@ -237,6 +237,18 @@ let expand_directory alt s =
                        (String.sub s 1 (String.length s - 1))
   else s
 
+let split_path_contents ?sep = function
+  | "" -> []
+  | s ->
+      let path_separator =
+        match sep, Sys.os_type with
+        | Some c, _ -> c
+        | None, ("Unix" | "Cygwin") -> ':'
+        | None, "Win32" -> ';'
+        | None, _ -> assert false
```

Comment:
Sorry for one more nit but `Sys.os_type` may end up being different from what you enumerated here (e.g. see here in [`ocaml-freestanding`](https://github.com/mirage/ocaml-freestanding/blob/41b7aa7d95d8c993cee7ad318be29ac9ebf2b497/configure.sh#L47)). I'm not sure `Misc.ml` could be used in such a context (it's only a runtime system being compiled there) but I think there's nothing bad about being robust to this opportunity and defaulting to e.g. `"Unix"` rather than fail with an assertion here.

Reply:
I don't *think* it is possible to build the compiler in a system where `os_type` is not one of the above, but the "default-is-unix" logic is present in other parts of the compiler, so I went ahead and amended as suggested.